app/src/test/java/org/greenstand/android/TreeTracker/splash/SplashScreenViewModelTest.kt

```diff
@@ -0,0 +1,144 @@
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import io.mockk.*
+import io.mockk.impl.annotations.MockK
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import org.greenstand.android.TreeTracker.MainCoroutineRule
+import org.greenstand.android.TreeTracker.analytics.ExceptionDataCollector
+import org.greenstand.android.TreeTracker.dashboard.TreesToSyncHelper
+import org.greenstand.android.TreeTracker.models.DeviceConfigUpdater
+import org.greenstand.android.TreeTracker.models.SessionTracker
+import org.greenstand.android.TreeTracker.models.UserRepo
+import org.greenstand.android.TreeTracker.models.location.LocationDataCapturer
+import org.greenstand.android.TreeTracker.models.messages.MessagesRepo
+import org.greenstand.android.TreeTracker.models.organization.OrgRepo
+import org.greenstand.android.TreeTracker.usecases.CheckForInternetUseCase
+import org.greenstand.android.TreeTracker.utils.FakeFileGenerator
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+@ExperimentalCoroutinesApi
+class SplashScreenViewModelTest {
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    private var orgJsonString : String? = null
+
+    @MockK(relaxed = true)
+    private lateinit var userRepo: UserRepo
+
+    @MockK(relaxed = true)
+    private lateinit var treesToSyncHelper: TreesToSyncHelper
+
+    @MockK(relaxed = true)
+    private lateinit var sessionTracker: SessionTracker
+
+    @MockK(relaxed = true)
+    private lateinit var deviceConfigUpdater: DeviceConfigUpdater
+
+    @MockK(relaxed = true)
+    private lateinit var locationDataCapturer: LocationDataCapturer
+
+    @MockK(relaxed = true)
+    private lateinit var messagesRepo: MessagesRepo
+
+    @MockK(relaxed = true)
+    private lateinit var checkForInternetUseCase: CheckForInternetUseCase
+
+    @MockK(relaxed = true)
+    private lateinit var orgRepo: OrgRepo
+
+    @MockK(relaxed = true)
+    private lateinit var exceptionDataCollector: ExceptionDataCollector
+
+    private lateinit var splashScreenViewModel: SplashScreenViewModel
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
+        splashScreenViewModel = SplashScreenViewModel(
+            orgJsonString = orgJsonString,
+            userRepo = userRepo,
+            treesToSyncHelper = treesToSyncHelper,
+            sessionTracker = sessionTracker,
+            deviceConfigUpdater = deviceConfigUpdater,
+            locationDataCapturer = locationDataCapturer,
+            messagesRepo = messagesRepo,
+            checkForInternetUseCase = checkForInternetUseCase,
+            orgRepo = orgRepo,
+            exceptionDataCollector = exceptionDataCollector
+        )
+    }
+
+    @Test
+    fun `all functions are executed when condition is true`() = runBlocking {
```

Comment:
Can you please update all the function names to be more descriptive about what they are testing?

Reply:
Fixed